src/orquestra/opt/optimizers/pso/continuous_pso_optimizer.py

```diff
@@ -0,0 +1,333 @@
+        topology_constructor : Callable[[int], SwarmTopology], optional
+            Class that receives the number of dimensions in the initialiser and
+            creates a SwarmTopology, by default StarTopology
+        seed : Optional[int], optional
+            Random seed for the numpy random number generator, by default None
+        recorder : RecorderFactory, optional
+            Recorder factory for keeping history of calls to the objective
+            function, by default _recorder
+        """
+        assert 0 < learning_rate <= 1.0, "Learning rate must be in (0, 1]."
+        assert 0 < inertia <= 1, "Inertia must be in (0, 1]"
+        assert (
+            affinity_towards_best_particle_position >= 0
+        ), "Affinity towards best particle position must be non-negative"
+        assert (
+            affinity_towards_best_swarm_position >= 0
+        ), "Affinity towards best swarm position must be non-negative"
+
+        # Convergence control:
+        assert (
+            patience is not None or max_iterations is not None or max_fevals is not None
+        ), (
+            "At least one of patience, max_iterations or max_fevals "
+            "must be specified to stop PSO"
+        )
+        super().__init__(recorder=recorder)
+
+        self.patience = patience
+        self.delta = delta
+        self.max_iterations = max_iterations
+        self.max_fevals = max_fevals
+
+        # Other attributes:
+        self.swarm_size = swarm_size
+        self.random_number_generator = np.random.default_rng(seed)
+        self.bounds = _get_bounds_like_array(bounds)
+        self.scale = self.bounds[1] - self.bounds[0]
+        self.shift = self.bounds[0]
+        self.function_at_best_positions = np.ones(swarm_size, dtype=float) * np.infty
+        self.inertia = inertia
+        self.topology_constructor = topology_constructor
+        self.affinity_towards_best_particle_position = (
+            affinity_towards_best_particle_position
+        )
+        self.affinity_towards_best_swarm_position = affinity_towards_best_swarm_position
+        self.learning_rate = learning_rate
+        self.velocity_bounds = velocity_bounds
+
+    def get_initial_velocities(self, dimensions: int) -> np.ndarray:
+        """
+        Initialises velocities for the particles in the swarm for a given number of
+        dimensions.
+
+        Parameters
+        ----------
+        dimensions : int
+            Number of dimensions of the problem.
+
+        Returns
+        -------
+        np.ndarray
+            (N, dimensions) array, where N is the swarm size.
+        """
+        velocities = self.random_number_generator.uniform(
+            0, 1, (self.swarm_size, dimensions)
+        )
+        if self.velocity_bounds is None:
+            self.velocity_bounds = None
+            velocities = 0.5 * (self.scale * velocities - self.shift)
+        else:
+            self.velocity_bounds = _get_bounds_like_array(self.velocity_bounds)
```

Comment:
[request] Could use similar refactor as I mentioned for "regular bounds"

Reply:
Not really because we need to store the velocity bounds. They are used elsewhere to clip the velocities.